src/io/mod.rs

```diff
@@ -7,3 +10,18 @@ pub use unix::IoDriver;
 
 #[cfg(not(target_family = "unix"))]
 compile_error!("Specified target platform is not supported (only unix family supported)");
+
+pub(crate) enum WritableData {
```

Comment:
You can add `len()` function to this enum

Reply:
I can't: offset is needed to produce this data, and we can't obtain offset without len. So len will be part of the creator trait

Reply:
I meant, that you can calculate it by the data within this enum:
```
impl WritableData {
    pub(crate) fn len(&self) -> usize {
         match self {
            // ....
         }
    }
}
```